### PR TITLE
Enhancement - Add viem support and remove ethers (NOT hw-wallets)

### PR DESCRIPTION
**Pull request by Adamj1232**
### Description
This PR removes Ethers and Bignumber.js from Core, common and injected and implements Viem support to handle balance checking and ENS details fetching. 
This resulted in reducing the bundle size of the core module from `861 kB` -> `459.8 kB` when minified!
Bundlephobia for [latest core](https://bundlephobia.com/package/@web3-onboard/core@2.21.6) & [viem core](https://bundlephobia.com/package/@web3-onboard/core@2.22.0-viem.2)

The packages are tagged with a `-viem.x` tag for release and testing in build envs.

So far common, core and injected have successfully published and I will test with them. There after I will incrementally add other packages to test with.

Any feedback would be very helpful

This will remain in draft until all packages have been tested and are ready for an `alpha` release

### CODE REVIEW

File: packages/common/src/utils.ts [hunk 1]
```diff
@@ -1,5 +1,81 @@
...
 
-export function weiToEth(wei: string): string {
-  return new Bignumber(wei).div(1e18).toString(10)
+const addressRegex = /^0x[a-fA-F0-9]{40}$/
+
+export const isAddress = (address: string): address is Address => {
+  return addressRegex.test(address)
+}
+
+export const weiHexToEth = (wei: string): string => {
```
Comment (aaronbarnardsound) [inline packages/common/src/utils.ts:9]: Does Viem have helper functions already for a lot of these?

File: packages/core/rollup.config.js [hunk 2]
```diff
@@ -5,14 +5,18 @@ import json from '@rollup/plugin-json'
 import sveltePreprocess from 'svelte-preprocess'
 import typescript from '@rollup/plugin-typescript'
 import copy from '@rollup-extras/plugin-copy'
+import commonjs from '@rollup/plugin-commonjs'
 
 const production = !process.env.ROLLUP_WATCH
+const specificPackages = ['idna-uts46-hx'] // Replace with the package names you want to target
```
Comment (aaronbarnardsound) [inline packages/core/rollup.config.js:11]: Remove this comment?

Comment (Adamj1232) [reply packages/core/rollup.config.js]: Was able to remove completely with the latest version of viem!

File: packages/common/src/utils.ts (hunk 1 above)
Comment (Adamj1232) [reply packages/common/src/utils.ts]: Yep updated to viem helpers which helps clean this up nicely!

**Comment by Adamj1232** [comment]
Closing as this will go out with wagmi support

**Pull request closed without merging**